runtime/lua/vim/treesitter/languagetree.lua

```diff
@@ -365,14 +365,166 @@ local function get_range_from_metadata(node, id, metadata)
   return { node:range() }
 end
 
+---@private
+---@param node TSNode
+---@param id integer
+---@param metadata TSMetadata
+---@return Range4[]
+local function get_node_ranges(node, id, metadata, include_children)
```

Comment:
We already have `vim.treesitter.get_node_range`-- can this be consolidated?

(May need a -- much needed -- cleanup of the node_range interface, so maybe defer and add a `TODO` here if it's out of scope?)

Reply:
Definitely out of scope. This was just to break the logic up.

Reply:
Could you add a TODO then?